project_group_setup.py

```diff
@@ -0,0 +1,270 @@
+
+import os
+import sys
+import getopt
+import comanage_scripts_utils as utils
+
+SCRIPT = os.path.basename(__file__)
+ENDPOINT = "https://registry-test.cilogon.org/registry/"
+LDAP_SERVER = "ldaps://ldap-test.cilogon.org"
+LDAP_USER = "uid=registry_user,ou=system,o=OSG,o=CO,dc=cilogon,dc=org"
+OSG_CO_ID = 8
+UNIX_CLUSTER_ID = 10
+LDAP_TARGET_ID = 9
+
+OSPOOL_PROJECT_PREFIX_STR = "Yes-"
+PROJECT_GIDS_START = 200000
+
+
+_usage = f"""\
+usage: [PASS=...] {SCRIPT} [OPTIONS]
+
+OPTIONS:
+  -u USER[:PASS]      specify USER and optionally PASS on command line
+  -c OSG_CO_ID        specify OSG CO ID (default = {OSG_CO_ID})
+  -g CLUSTER_ID       specify UNIX Cluster ID (default = {UNIX_CLUSTER_ID})
+  -l LDAP_TARGET      specify LDAP Provsion ID (defult = {LDAP_TARGET_ID})
+  -p LDAP authtok     specify LDAP server authtok
+  -d passfd           specify open fd to read PASS
+  -f passfile         specify path to file to open and read PASS
+  -e ENDPOINT         specify REST endpoint
+                        (default = {ENDPOINT})
+  -o outfile          specify output file (default: write to stdout)
+  -h                  display this help text
+
+PASS for USER is taken from the first of:
+  1. -u USER:PASS
+  2. -d passfd (read from fd)
+  3. -f passfile (read from file)
+  4. read from $PASS env var
+"""
+
+
+def usage(msg=None):
+    if msg:
+        print(msg + "\n", file=sys.stderr)
+
+    print(_usage, file=sys.stderr)
+    sys.exit()
+
+
+class Options:
+    endpoint = ENDPOINT
+    user = "co_8.william_test"
+    osg_co_id = OSG_CO_ID
+    ucid = UNIX_CLUSTER_ID
+    provision_target = LDAP_TARGET_ID
+    outfile = None
+    authstr = None
+    ldap_authtok = None
+    project_gid_startval = PROJECT_GIDS_START
+
+
+options = Options()
+
+
+def parse_options(args):
+    try:
+        ops, args = getopt.getopt(args, "u:c:g:l:p:d:f:e:o:h")
+    except getopt.GetoptError:
+        usage()
+
+    if args:
+        usage("Extra arguments: %s" % repr(args))
+
+    passfd = None
+    passfile = None
+
+    for op, arg in ops:
+        if op == "-h":
+            usage()
+        if op == "-u":
+            options.user = arg
+        if op == "-c":
+            options.osg_co_id = int(arg)
+        if op == "-g":
+            options.ucid = int(arg)
+        if op == "-l":
+            options.provision_target = int(arg)
+        if op == "-p":
+            options.ldap_authtok = arg
+        if op == "-d":
+            passfd = int(arg)
+        if op == "-f":
+            passfile = arg
+        if op == "-e":
+            options.endpoint = arg
+        if op == "-o":
+            options.outfile = arg
+
+    try:
+        user, passwd = utils.getpw(options.user, passfd, passfile)
+        options.authstr = utils.mkauthstr(user, passwd)
+    except PermissionError:
+        usage("PASS required")
+
+
+def append_if_project(project_groups, group):
+    # If this group has a ospoolproject id, and it starts with "Yes-", it's a project
```

Comment:
```suggestion
    """If this group has a ospoolproject id, and it starts with "Yes-", it's a project"""
```

May as well use docstrings as they show up nicely in the Python interpreter